onnxruntime/core/optimizer/layout_transformation/layout_transformation.cc

```diff
@@ -177,7 +177,11 @@ Status TransformLayoutForEP(Graph& graph, bool& modified, const IExecutionProvid
         for (size_t i = 2; i < node->Inputs().size(); i++) {
           auto constant = api_graph->GetConstant(node->Inputs()[i]);
           if (constant != nullptr && constant->Data().size() > 0) {
-            input_perms.push_back(&input_perm);
+            // Starting from opset version 18, 'scales' and 'sizes' can be 2D tensors.
+            // However, our current implementation only supports the transposition of 4D tensors.
+            if (constant->NumElements() == 4) {
+              input_perms.push_back(&input_perm);
+            }
```

Comment:
If there are 2 entries in axes can we not handle that? i.e. user specifies just the H and W scale/size.

Reply:
When `axes` is provided, `scales` and `sizes` are 2D tensors. We can just leave it to EPs to handle that.

Reply:
Ah right. If there's no N or C values we don't need to shuffle anything.

nit: from opset 18 on I believe `scales` and `sizes` can be any length up to the input rank, so 'can be 2D tensors' is slightly misleading.

Reply:
> nit: from opset 18 on I believe scales and sizes can be any length up to the input rank, so 'can be 2D tensors' is slightly misleading.

Indeed, fixed.